ffxiv_visland/Gathering/GatherRouteExec.cs

```diff
@@ -67,7 +67,9 @@ public unsafe void Update()
 
         var wp = CurrentRoute.Waypoints[CurrentWaypoint];
         var toWaypoint = wp.Position - Player.Object.Position;
-        var needToGetCloser = toWaypoint.LengthSquared() > wp.Radius * wp.Radius;
+        // a target of exactly 0 means "do this without moving"
+        var needToGetCloser = wp.Position != Vector3.Zero && toWaypoint.LengthSquared() > wp.Radius * wp.Radius;
```

Comment:
Since I'm using lay of the land abilities my flags are off the object table. I assume this check is because yours can be too but that's surprising. Anyway what I think is a better solution (and what I did) was check in the beginning of the update loop for nodes that should be your node and try to update the exact positions and other info along the way to the node.

Reply:
The point is more to allow position-neutral waypoints. There are many possible actions where you don't care about the position, or worse, it's entirely inconvenient to have it trying to reach a certain position in a certain zone.

Take the `UseAction` interaction type, for example. If I'm doing something like toggling Triangulate, I really don't want to be going somewhere to do it. I just want to use the action *where I am*. This also makes the `StartRoute` interaction more usable as you can have generic routes that aren't based on a certain map.

The demo is entirely position neutral via this logic, it can be run literally anywhere and starts based on your current position. 
![image](https://github.com/user-attachments/assets/0e85fc5e-9b46-4204-a2f4-85fd910cc652)

Reply:
Oh, I get it. You were the one in the server that asked about that too. Yeah this is a decent enough hack, I'll add it in

Reply:
> You were the one in the server that asked about that too.
Nope, not in any dev servers